### ISSUE TITLE
AI use insta-fliсker on Grief when player hasn't cards in hands

### ISSUE THREAD

**Issue by MorihiroS** [Card-Forge/forge#5891]
**Describe the bug**
AI casts Ephemerate on Grief, which has been on the battlefield for a long time, when player hasn't cards in hands

**To Reproduce**
1. AI has Grief on the battlefield and Ephemerate in hands
2. Player hasn't cards in hands
3. AI casts Ephemerate and targets Grief

**Expected behavior**
AI should cast Ephemerate on Grief when player has cards in hands (and before evoke trigger)

**Additional context**
Similar with #4233

**Desktop**
OS: Windows 10
Java Version: 1.8.0_202-b08
Forge Version: 1.6.64-SNAPSHOT-08.04

**Screenshots**

![grief1](https://github.com/user-attachments/assets/4ae6a155-3759-49b8-a092-571a14ea162b)
![grief2](https://github.com/user-attachments/assets/d92a313f-3e65-4d23-8c9e-fd919fdd96f1)

**Comment by MorihiroS** [comment]
@Hanmac, need AI label, keep label and assign someone, please

**Comment by MorihiroS** [comment]
@Hanmac, need AI label, keep label and assign someone, please

**Comment by Hanmac** [comment]
it depends on if the creature was Evoked or not

flickering an Evoked Grief might be useful, even if opponent has no cards in hand

**Comment by MorihiroS** [comment]
> flickering an Evoked Grief might be useful, even if opponent has no cards in hand

@Hanmac, Grief didn't even have time to get out of summoning sickness and go on the attack, there was no point in that action, neither for untapping nor for defense

**Comment by Hanmac** [comment]
Flickering and Evoked Grief prevents it from being sacrificed

**Comment by MorihiroS** [comment]
@Hanmac, yes, but in the screenshots, the player was doing absolutely nothing...

**Comment by MorihiroS** [comment]
> Flickering and Evoked Grief prevents it from being sacrificed

@Hanmac, yes, but in the screenshot it was cast for mana, there was no sacrifice trigger.

**Comment by MorihiroS** [comment]
> Flickering and Evoked Grief prevents it from being sacrificed

@Hanmac, yes, but in the screenshot it was cast for mana, there was no sacrifice trigger.
